apps/devhub/tests/test_views.py

```diff
@@ -2483,6 +2483,45 @@ def test_unlisted_addon_pass_validation_with_flag(self, mock_sign_file):
         assert file_.status == amo.STATUS_LITE
         assert mock_sign_file.called
 
+    @mock.patch('devhub.views.sign_file')
+    def test_beta_addon_pass_validation_with_flag(self, mock_sign_file):
+        """Beta files that pass validation are automatically signed/reviewed if
+        the flag is enabled."""
+        self.create_flag('automatic-validation')
+        version = self.addon.latest_version
+        version.all_files[0].update(status=amo.STATUS_BETA)
+        # Make sure the file has no validation warnings nor errors.
+        self.upload.update(
+            validation='{"notices": 2, "errors": 0, "messages": [],'
+                       ' "metadata": {}, "warnings": 0}')
+        assert self.addon.status == amo.STATUS_PUBLIC
+        # With the flag: should be signed.
+        self.post(beta=True)
+        file_ = File.objects.all().order_by('-created')[0]
+        # File status didn't change and the file is signed.
+        assert self.addon.reload().status == amo.STATUS_PUBLIC
+        assert file_.status == amo.STATUS_BETA
+        assert mock_sign_file.called
+
+    @mock.patch('devhub.views.sign_file')
+    def test_beta_addon_not_passed_validation_with_flag(self, mock_sign_file):
+        """Beta files that don't pass validaiton should raise an error."""
+        self.create_flag('automatic-validation')
+        version = self.addon.latest_version
+        version.all_files[0].update(status=amo.STATUS_BETA)
+        # Make sure the file has validation warnings.
+        self.upload.update(
+            validation='{"notices": 2, "errors": 1, "messages": [],'
+                       ' "metadata": {}, "warnings": 1}')
+        assert self.addon.status == amo.STATUS_PUBLIC
+        response = self.post(beta=True)
```

Comment:
you need `self.post(beta=True, expected_status=403)`, and you can remove the following line.